src/components/scenegraph/Toolbar.js

```diff
@@ -148,9 +181,60 @@ export default class Toolbar extends React.Component {
     let objects = this.state.objects;
     let objectChanges = [];
     let changedObjectsString = "";
-    for(var id in AFRAME.INSPECTOR.history.updates)
-      objectChanges.push([parseInt(id.replace("-obj", "")), AFRAME.INSPECTOR.history.updates[id]]);
+
+    for(var id in AFRAME.INSPECTOR.history.updates){
+      if (id.endsWith("-obj")){
+        objectChanges.push([parseInt(id.replace("-obj", "")), AFRAME.INSPECTOR.history.updates[id]]);
+      } else {
+        const objName = id.split("@")[0];
+        if (!('gltf-model' in AFRAME.INSPECTOR.history.updates[id]) || AFRAME.INSPECTOR.history.updates[id]['gltf-model'] == ""){
+          alert("Error: Save failed, Please provide gltf-model for object with name: " + id);
+          return;
+        } else{
+          let basicObject = {
+            "position": [0.0, 0.0, 0.0],
+            "scale": [1.0, 1.0, 1.0],
+            "rotation": [0.0, 0.0, 0.0],
+            "name": objName,
+            "asset_id": 1,
+            "next_objects": [
+              {
+                "id": 2,
+                "action": {
+                  "type": "text",
+                  "text_id": 1
+                }
+              }
+            ],
+            "is_interactable": false
+          }
+          let curChanges = AFRAME.INSPECTOR.history.updates[id];
+          for (const prop in curChanges){
+            if (prop == "position" || prop == "scale" || prop == "rotation"){
+              basicObject[prop] = curChanges[prop].split(" ").map(Number);
+            } else if (prop == "gltf-model"){
+              basicObject["asset_id"] = this.state.linkToIdMap[curChanges[prop]];
```

Comment:
Do we need `linkToIdMap` here?

Reply:
Yes, otherwise we don't know what the asset_id is